examples/progress-bar/a-lot-of-parallel-tasks.py

```diff
@@ -16,8 +16,28 @@ def main():
         bottom_toolbar=HTML("<b>[Control-L]</b> clear  <b>[Control-C]</b> abort"),
     ) as pb:
 
+        lock = threading.Lock()
+        completed = []
+
+        # Remove a completed counter after 5 seconds but keep the last 5.
```

Comment:
This is nice. That's a great use case.

I'm still not convinced though that a thread is the right approach here. We will get many threads that never finish and can't be joined anywhere. I'm thinking about whether we could reuse the prompt_toolkit filters in this case too, and do something like:
```python
@Condition
def ten_seconds_passed() -> bool:
    counter = get_counter()  # context local
    return time.time() - counter.stop_time > 10

@Condition
def at_least_five_bars() -> bool:
    pb = get_progress_bar()  # context local
    return len(pb.counters) > 10

for i in pb(range(total), label=label, remove_when_done=ten_seconds_passed & at_least_five_bars):
```

If this condition evaluates  during every render, we can remove the bar when this turns False.